app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -72,23 +70,55 @@ class DatabaseCleanupManagerImpl(
         launch {
             val configuration = configurationRepository.observeConfiguration().blockingFirst()
 
-            val thresholdRedMessages = ZonedDateTime.now().minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Red, thresholdRedMessages)
-
-            val thresholdYellowMessages = ZonedDateTime.now().minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Yellow, thresholdYellowMessages)
-
-            val thresholdGreenMessages = ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.Revoke.Suspicion, thresholdGreenMessages)
+            val thresholdRedMessages = ZonedDateTime.now()
+                .minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.InfectionLevel.Red,
+                thresholdRedMessages
+            )
+
+            val thresholdYellowMessages = ZonedDateTime.now()
+                .minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.InfectionLevel.Yellow,
+                thresholdYellowMessages
+            )
+
+            val thresholdGreenMessages =
+                ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
```

Comment:
THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES should also be backend controlled.

Reply:
This line has been brought in the review because it was formatted, sorry for that. 

This part touches the download functionality of the app, i would not update it since will suffer some changes in the implementation for download.